### PR TITLE
Support history as default view for filemanager

### LINKED ISSUE #2774: support history as the default view for filemanager
see discussion in https://github.com/koreader/koreader/issues/2614

This should be a configurable thing.

### PR DESCRIPTION
**Pull request by Frenzie**
Fixes #2774

* make history view borderless
* add new menu for what to start with

# Before
screenshot_2017-08-08_12-02-33

screenshot_2017-08-08_12-15-21

# After
screenshot_2017-08-08_12-01-37

screenshot_2017-08-08_12-01-45

screenshot_2017-08-08_12-01-51

# Take note

I didn't bother to implement the "feature" that you couldn't toggle this setting unless you opened a document first. Comments welcome @ersi @KenMaltby

screenshot_2017-08-08_12-03-40

### CODE REVIEW

**Comment by KenMaltby** [comment]
If the "Start with history >" brings up the menu to select between the three start up options, then it might be better named "Start up options >", for one thing it wouldn't always be set on history.  Not sure the old option, shown greyed out but checked, in the last screenshot, would be needed anymore.

**Comment by Frenzie** [comment]
> If the "Start with history >" brings up the menu to select between the three start up options, then it might be better named "Start up options >"

It displays the current setting. I could change it to `Start with: blabla`

**Comment by Frenzie** [comment]
Screenshot of the result of the commit I just pushed:

screenshot_2017-08-13_20-54-07

File: frontend/apps/filemanager/filemanager.lua [hunk 1]
```diff
@@ -483,6 +483,40 @@ function FileManager:getSortingMenuTable()
...
+    }
+    local set_sw_table = function(start_with)
+        return {
+            text = start_withs[start_with][2],
+            checked_func = function()
+                return start_with_setting == start_with
+            end,
+            callback = function()
+                start_with_setting = start_with
+                G_reader_settings:saveSetting("start_with", start_with)
```
Comment (Frenzie) [inline frontend/apps/filemanager/filemanager.lua]: @Hzj-jie Should there be a flush after this or is it fine this way?

**Comment by poire-z** [comment]
Noticed something strange, both on emulator and my kobo:
When you go to History from a menu, it displays full screen like on your screenshot.
But when you choose "Start with: history", the first opening on startup is not full screen, there's the top title from FileBrowser :
<kbd>filemanager_2017-aug-17_215518</kbd>
Subsequent openings of History are fine and full screen.

**Comment by poire-z** [comment]
Oups, it seems it is a side effect of my coverbrowser plugin :(
History displays nicely on startup if i move out the plugins/coverbrowser.koplugin/ directory...
I'll look at that tomorrow.

**Comment by poire-z** [comment]
The following seems to solve the problem (with display modes remaining `classic`, and with the other modes). But nextTicking a nextTick looks strange (but better than a scheduleIn with an arbitrary delay).
```lua
diff --git a/reader.lua b/reader.lua
index 664e0486..fcfabc2b 100755
--- a/reader.lua
+++ b/reader.lua
@@ -171,7 +171,11 @@ if ARGV[argidx] and ARGV[argidx] ~= "" then
         if start_with == "history" then
             local FileManagerHistory = require("apps/filemanager/filemanagerhistory")
             UIManager:nextTick(function()
-                FileManagerHistory:onShowHist(last_file)
+                -- we need to postpone it a tick so that coverbrowser
+                -- may do its own setup/sizing
+                UIManager:nextTick(function()
+                    FileManagerHistory:onShowHist(last_file)
+                end)
             end)
         end
     end

```
What do you think ?

**Comment by Frenzie** [comment]
That does look slightly odd but nothing more than that. Does it help at all if you put it inside the "main" nextTick()?

https://github.com/koreader/koreader/blob/3b5cd4c23b53ade1682699b8ff082fab7b23c350/reader.lua#L166-L168

**Comment by poire-z** [comment]
Yes it does, it's actually smoother (quicker refresh when displaying history over the original file browser view) :
```lua
        UIManager:nextTick(function()
            FileManager:showFiles(home_dir)
            -- always open history on top of filemanager so closing history
            -- doesn't result in exit
            if start_with == "history" then
                local FileManagerHistory = require("apps/filemanager/filemanagerhistory")
                UIManager:nextTick(function()
                    FileManagerHistory:onShowHist(last_file)
                end)
            end
        end)
```
There is no noticable refresh when there is no coverbrowser.kolugin, but well, as I wanted to have it self contained with no core modification, I needed to do an update after the original core stuff is set up.
I'll check again tomorrow if I can avoid that for classic mode.

**Comment by Frenzie** [comment]
Alright, so let's go with that?

**Comment by poire-z** [comment]
No, I found a better fix : removing the CenterContainer in FileManagerHistory:onShowHist() , which is no more needed as it is now full screen. No alignemnt problem after that.
And I will let reader.lua as it is, as the nested nextTick makes for the little flashing. With your original reader.lua, and the FileManagerHistory fix, "start with history" is quick, smooth and with no flashing.

**Pull request merged**